### PR TITLE
Attempt to standardize class guards for enterprise hub policy

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom please test this in jenkins (define am_policy_hub in one test)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7658/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7658/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7658/

**Comment by nickanderson** [comment]
@cf-bottom please test this in jenkins (define am_policy_hub and run with verbose in one test)

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7659/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7659/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7659/

**Comment by nickanderson** [comment]
@cf-bottom jenkins please.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7661/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7661/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7661/

**Comment by nickanderson** [comment]
@cf-bottom jenkins please.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7665/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7665/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7665/

**Comment by olehermanse** [comment]
@cf-bottom please test this in jenkins :)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7666/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7666/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7666/

**Comment by olehermanse** [comment]
@cf-bottom please test in jenkins, again.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7667/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7667/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7667/

File: promises.cf.in [hunk 1]
```diff
@@ -213,7 +213,7 @@ bundle common cfe_internal_inputs
         comment => "This policy configures external watchdogs to ensure that
                     cf-execd is always running.";
 
-    enterprise_edition.(policy_server|am_policy_hub)::
+    any::
```
Comment (nickanderson) [inline promises.cf.in:216]: This change seemed to have a positive effect on the test outcome

Comment (olehermanse) [reply promises.cf.in]: I wonder if we can undo this and add class guards to the methods promises instead.

**Comment by olehermanse** [comment]
@cf-bottom I've tried to add the same class guards to several methods promises. Please test this in Jenkins.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7671/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7671/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7671/

**Review by nickanderson — APPROVED** [review]

**Comment by olehermanse** [comment]
Relates to this change:
https://github.com/cfengine/core/pull/4929

But can be merged alone.

**Pull request merged**